Shopify/shopify-cli
# Feature: Intallation of shopify-cli downloads shopify-extensions binary

### WHY are these changes introduced?

[Enables Development Server](https://github.com/Shopify/shopify-cli-extensions/issues/238) for development of checkout extensions.

### WHAT is this pull request doing?

Installation of the `shopify-cli` gem will trigger download of `shopify-extensions` binary.

https://user-images.githubusercontent.com/4490738/157476748-5a2fadfd-82ed-44f4-8a10-1636c8cb93ac.mp4

### How to test your changes?

1. Navigate to `Shopify/shopify-cli` directory.
2. From the CLI, run `MY_TMP_DIR=$(ruby -e 'puts Gem.dir')` and to save off the target gem installation directory. If using multiple shell sessions you will need to `export MY_TMP_DIR`.
3. Checkout this branch: `git fetch; git checkout origin/checkout-extensions-support`
4. Run `rake install` to build the gem locally to `pkg/`. You can verify this with `ls pkg/`. The next steps assume the output file is `shopify-cli-2.13.0.gem` but adjust your version as needed.
5. Run `gem install --local pkg/shopify-cli-2.13.0.gem` to install the gem built in step 4 to the system directory from step 2 (`MY_TMP_DIR`).
6. Navigate to the local gem installation extension directory: `cd $MY_TMP_DIR/gems/shopify-cli-2.13.0/ext/shopify-extensions`. 
7. Run `./shopify-extensions version` and verify the output is `v0.2.0`
8. Unset the temporary variable: `unset MY_TMP_DIR`

### Post-release steps



### Update checklist
